components/eamxx/src/share/atm_process/atmosphere_process_group.cpp

```diff
@@ -149,6 +149,24 @@ AtmosphereProcessGroup (const ekat::Comm& comm, const ekat::ParameterList& param
   }
 }
 
+std::shared_ptr<AtmosphereProcessGroup::atm_proc_type>
+AtmosphereProcessGroup::get_process_nonconst (const std::string& name) const {
+  EKAT_REQUIRE_MSG(has_process(name), "Error! Process "+name+" requested from group "+this->name()+
+                                      " but is not constained in this group.\n");
+
+  std::shared_ptr<atm_proc_type> return_process;
+  for (auto& process: m_atm_processes) {
+    if (process->type() == AtmosphereProcessType::Group) {
+      const auto* group = dynamic_cast<const AtmosphereProcessGroup*>(process.get());
+      return group->get_process_nonconst(name);
+    } else if (process->name() == name) {
+        return_process = process;
+        break;
+    }
+  }
+  return return_process;
+}
```

Comment:
New helper function for finding a field by name. Could just implement these lines in the driver where this function is called if we don't want this version.

Reply:
I think it's fine.